src/redis-brain.coffee

```diff
@@ -34,9 +35,13 @@ module.exports = (robot) ->
     robot.logger.info "Using default redis on localhost:6379"
 
 
-  info   = Url.parse  redisUrl, true
-  client = Redis.createClient(info.port, info.hostname)
-  prefix = info.path?.replace('/', '') or 'hubot'
+  info = Url.parse  redisUrl, true
+  if info.hostname == ''
+    client = Redis.createClient(info.pathname)
+    prefix = info.query?.toString() or 'hubot'
+  else
+    client = Redis.createClient(info.port, info.hostname)
+    prefix = info.pathname?.replace('/', '') or info.query?.toString() or 'hubot'
```

Comment:
It seems that `info.pathname` is always `'/'` even if not given:

```
coffee> url.parse('https://foo').pathname
'/'
```

The replace would make it `''`, but that is considered truthy. That means the second part of this expression would never actually get used.